docs/code/CrossPlatform.md

```diff
@@ -0,0 +1,157 @@
+# Cross-Platform and Architecture design doc
+
+### Table of contents  <!-- omit in toc -->
+- [Cross-Platform and Architecture design doc](#cross-platform-and-architecture-design-doc)
+	- [1. Why cross-platform/architecture](#1-why-cross-platformarchitecture)
+	- [2. Current status](#2-current-status)
+		- [2.1 Problems](#21-problems)
+		- [2.2 Build](#22-build)
+		- [2.3 Managed Code](#23-managed-code)
+		- [2.4 Native Projects](#24-native-projects)
+		- [2.5 3rd Party Dependencies](#25-3rd-party-dependencies)
+	- [3 Possible Solutions](#3-possible-solutions)
+		- [3.1 Fully Remove Intel MKL](#31-fully-remove-intel-mkl)
+		- [3.2 Find Replacements as needed](#32-find-replacements-as-needed)
+		- [3.3 Software fallbacks](#33-software-fallbacks)
+		- [3.4 Hybrid Between Finding Replacements and Software Fallbacks](#34-hybrid-between-finding-replacements-and-software-fallbacks)
+	- [My Suggestion](#my-suggestion)
+		- [NET Core 5](#net-core-5)
+		- [3rd Party Dependencies](#3rd-party-dependencies)
+		- [Helix](#helix)
+		- [Mobile Support](#mobile-support)
+		- [Support Grid](#support-grid)
+
+## 1. Why cross-platform/architecture
+ML.NET is an open-source machine learning framework which makes machine learning accessible to .NET developers with the same code that powers machine learning across many Microsoft products, including Power BI, Windows Defender, and Azure.
+
+ML.NET allows .NET developers to develop/train their own models and infuse custom machine learning into their applications using .NET.
+
+Currently, while .NET is able to run on many different platforms and architectures, ML.NET is only able to run on Windows, Mac, and Linux, either x86 or x64. This excludes many architectures such as Arm, Arm64, M1, and web assembly, places that .NET is currently able to run.
+
+The goal is to enable ML.NET to run everywhere that .NET itself is able to run.
+
+## 2. Current status
+### 2.1 Problems
+ML.NET has a hard dependency on Intel MKL. It performs many optimized math functions and enables several transformers/trainers to be run natively for improved performance. The problem is that Intel MKL can only run on x86/x64 machines and is the main blocker for expanding to other architectures.
```

Comment:
```suggestion
Parts of ML.NET have a hard dependency on Intel MKL. It performs many optimized math functions and enables several transformers/trainers to be run natively for improved performance. The problem is that Intel MKL can only run on x86/x64 machines and is the main blocker for expanding to other architectures.
```

The current wording makes it sound like all of ML.NET needs MKL. But in reality only one trainer and 2 transforms do (IIRC).

Reply:
Let me figure out the exact trainers/transforms that do. Even if its only 2 of each I would still say its a hard dependency for 2 main reasons:
 - The build process itself requires it. If Intel MKL is not present, the native components of the build will fail.
 - The base test class tries to setup Intel MKL in its constructor, even for tests that don't use it at all.

With these 2 things, you can't build or test without Intel MKL present, even if you are not using anything that needs it.

Thoughts?